Code/src/main/java/nl/utwente/group10/ui/components/blocks/FunctionBlock.java

```diff
@@ -327,4 +352,20 @@ public final void error() {
     public OutputAnchor getOutputAnchor() {
         return output;
     }
+
+    @Override
+    public void changed(ObservableValue<? extends Number> arg0, Number arg1, Number arg2) {
+        if (arg0.equals(bowtieIndex)) {
+            invalidateBowtieIndex();
+        }
+    }
+
+    private void invalidateBowtieIndex() {
+        for (InputAnchor input : getAllInputs()) {
+            input.setVisible(getInputIndex(input) < getBowtieIndex());
+            if(input.isConnected()){
```

Comment:
Owja dit is wat onhandig, ik had het nog niet getest met op runtime de bowtie index aanpassen.